tui/menu.go

```diff
@@ -101,7 +104,24 @@ func newMenuPage(tui *Tui) (Page, error) {
 
 	page.installBtn = CreateSimpleButton(page.cFrame, AutoSize, AutoSize, "Install", Fixed)
 	page.installBtn.OnClick(func(ev clui.Event) {
-		page.GotoPage(TuiPageInstall)
+		if controller.NetworkPassing {
+			// Network already validated, just start the install
+			page.GotoPage(TuiPageInstall)
+		} else {
+			// Network needs to be validated before the install
+			if dialog, err := CreateNetworkTestDialogBox(page.tui.model); err == nil {
+				if dialog.RunNetworkTest() {
+					// Automatically close if it worked
+					clui.RefreshScreen()
+					time.Sleep(time.Second)
```

Comment:
Do you really need this sleep here?

Reply:
It was only there to give the user a "second" to see the Success status before the dialog automatically goes away. I added it because it bothered me that the popup went away too quickly. If you have a strong feeling about this, we can remove it.